### PR TITLE
CORCI-768 build: Update for repository names.

### PR DESCRIPTION
**Pull request by JohnMalmberg**
The simplified repository names just use sles12 and leap42 in their
names.

Signed-off-by: John E Malmberg <john.e.malmberg@intel.com>

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -79,7 +79,7 @@ pipeline {
                             publishToRepository product: 'slurm',
                                                 format: 'yum',
                                                 maturity: 'stable',
-                                                tech: 'sles12.3',
+                                                tech: 'sles-12',
```
Comment (brianjmurrell) [inline Jenkinsfile]: Don't we have a pretty well established convention of not using a `-` between the distro name and version?  Is there a particular reason to deviate here?

File: Jenkinsfile [hunk 2]
```diff
@@ -123,7 +123,7 @@ pipeline {
                             publishToRepository product: 'slurm',
                                                 format: 'yum',
                                                 maturity: 'stable',
-                                                tech: 'leap42.3',
+                                                tech: 'leap-42',
```
Comment (brianjmurrell) [inline Jenkinsfile]: Ditto.

File: Jenkinsfile (hunk 1 above)
Comment (JohnMalmberg) [reply Jenkinsfile]: This was done to make the repository names more readable and more easily parsed by scripts maintaining Nexus server, especially when we need to group by distro and then by version.

**Review by jolivier23 — APPROVED** [review]
LGTM.   Though if we are deviating from a standard practice, we should have a good reason.

**Comment by JohnMalmberg** [comment]
A sampling of the upstream repository URLs show that they are generally separating the distro from the version in the path name with some type of delimiter such as a slash, dash, or underscore.
/epel/7
https://copr-be.cloud.fedoraproject.org/results/jhli/ipmctl/epel-7-x86_64/
http://mirror.centos.org/centos/7.6.1810/extras/x86_64/
https://download.opensuse.org/repositories/openSUSE:Tools/openSUSE_42.3/

**Review by brianjmurrell — APPROVED** [review]

**Review by brianjmurrell — CHANGES_REQUESTED** [review]
Why isn't this using the packaging repo and linting `Jenkinsfile` stages?

**Comment by JohnMalmberg** [comment]
I did not take the time to make sure all of that worked for this.  When SLES-12sp3 usage is retired, this repository will no longer be used.

I just needed the changes to save the artifacts to the repository until that point, as my priority is getting SLURM running in CI.

**Comment by brianjmurrell** [comment]
> I did not take the time to make sure all of that worked for this.

Fair enough.  It's in #7 and passing tests.  We should just merge that to here and then land this.

**Review by JohnMalmberg — COMMENTED** [review]
Approved.
I can not check the box since I originated the PR.

**Review by brianjmurrell — APPROVED** [review]

**Pull request merged**